packages/statemanager/src/ethersStateManager.ts

```diff
@@ -109,24 +109,22 @@ export class EthersStateManager implements StateManagerInterface {
    */
   async getContractStorage(address: Address, key: Uint8Array): Promise<Uint8Array> {
     // Check storage slot in cache
-    const accountStorage: Map<string, Uint8Array> | undefined = this.storageCache.get(
-      address.toString()
-    )
-    let storage: Uint8Array | string | undefined
-    if (accountStorage !== undefined) {
-      storage = accountStorage.get(bytesToHex(key))
-      if (storage !== undefined) {
-        return storage
-      }
+    if (key.length !== 32) {
+      throw new Error('Storage key must be 32 bytes long')
+    }
+
+    let value = this.storageCache!.get(address, key)
+    if (value !== undefined) {
+      return value
     }
 
     // Retrieve storage slot from provider if not found in cache
-    storage = await this.provider.getStorageAt(
+    const storage = await this.provider.getStorageAt(
       address.toString(),
       bytesToBigInt(key),
       this.blockTag
     )
-    const value = toBytes(storage)
+    value = toBytes(storage)
```

Comment:
Should it also put the value in the storage here?

Reply:
It does two lines below

Reply:
Ah ok great :+1: